core/src/main/java/org/wso2/msf4j/internal/MicroservicesServerSC.java

```diff
@@ -102,12 +186,28 @@ protected void removeInterceptor(Interceptor interceptor) {
             policy = ReferencePolicy.DYNAMIC,
             unbind = "removeExceptionMapper"
     )
-    protected void addExceptionMapper(ExceptionMapper exceptionMapper) {
-        microservicesRegistry.addExceptionMapper(exceptionMapper);
+    protected void addExceptionMapper(ExceptionMapper exceptionMapper, Map properties) {
+        Object channelId = properties.get(MSF4JConstants.CHANNEL_ID);
+        Map<String, MicroservicesRegistry> microservicesRegistries =
+                DataHolder.getInstance().getMicroservicesRegistries();
+        if (channelId != null) {
+            MicroservicesRegistry microservicesRegistry = microservicesRegistries.get(channelId.toString());
+            if (microservicesRegistry == null) {
+                throw new RuntimeException("Couldn't found the registry for channel ID " + channelId);
+            }
+            microservicesRegistry.addExceptionMapper(exceptionMapper);
+        } else {
+            microservicesRegistries.values().forEach(registry -> registry.addExceptionMapper(exceptionMapper));
+        }
     }
 
-    protected void removeExceptionMapper(ExceptionMapper exceptionMapper) {
-        microservicesRegistry.removeExceptionMapper(exceptionMapper);
+    protected void removeExceptionMapper(ExceptionMapper exceptionMapper, Map properties) {
+        Object channelId = properties.get(MSF4JConstants.CHANNEL_ID);
+        Map<String, MicroservicesRegistry> microservicesRegistries =
+                DataHolder.getInstance().getMicroservicesRegistries();
+        if (channelId != null) {
+            microservicesRegistries.get(channelId.toString()).removeExceptionMapper(exceptionMapper);
+        }
```

Comment:
Need else condition here and remove from all microservices registries, just like you did when you added to all microservices registries. Need to do the same for services, interceptors etc.